src/graphics.c

```diff
@@ -2024,23 +2024,47 @@ GpStatus WINGDIPAPI
 GdipSetClipPath (GpGraphics *graphics, GpPath *path, CombineMode combineMode)
 {
 	GpStatus status;
+	GpPath *work;
 
 	if (!graphics || !path)
 		return InvalidParameter;
 
-	status = GdipCombineRegionPath (graphics->clip, path, combineMode);	
+	/* if the matrix is empty, avoid path cloning and transform */
+	if (gdip_is_matrix_empty (graphics->clip_matrix)) {
+		work = path;
+	} else {
+		cairo_matrix_t inverted;
+
+		gdip_cairo_matrix_copy (&inverted, graphics->clip_matrix);
+		cairo_matrix_invert (&inverted);
+
```

Comment:
I actually discovered that there's always an inverted version of `graphics->clip_matrix` in `graphics->copy_of_ctm`. I'd prefer to simplify the code in a separate patch though, along with existing code in `GdipSetClipRect` and `GdipSetClipRegion`.